src/components/button/button.scss

```diff
@@ -5,11 +5,14 @@
 @import "./../../style-config/buttons";
 
 .carbon-button {
+  align-items: center;
   border: $carbon-button__border--secondary;
   border-radius: $carbon-button__border-radius;
   box-sizing: border-box;
   cursor: pointer;
-  display: inline-block;
+  display: inline-flex;
```

Comment:
The screenshots are from the demo the site in Safari. To see the hamburger menu you need to shrink the browser a bit.